### PR TITLE
Dwc0011/fix codecommit error

### LINKED ISSUE #4848: Unexpected repo provider 'codecommit' error
Codecommit repos are not being handled resulting in "Unexpected repo provider 'codecommit'" errors.

After searching the repo there are multiple places that still need codecommit logic to be addressed.

You can view conditionals that need to include codecommit [here](https://github.com/dependabot/dependabot-core/search?q=%22when+%5C%22bitbucket%5C%22%22)

As in previous Issues I will begin a PR and start addressing these issue.

**Example error received:**
/dependabot-common-0.175.0/lib/dependabot/metadata_finders/base/changelog_finder.rb:223:in `fetch_dependency_file_list': Unexpected repo provider 'codecommit' (RuntimeError)

### PR DESCRIPTION
**Pull request by dwc0011**
Fixes #4848

Adds codecommit handling to each conditional where needed.

Adds TODO messages for future work needed.  This PR is addresses the error preventing the PR from being created only.  

Live run tests have been performed against a codecommit projects and sources and were able to be created properly.

### CODE REVIEW

**Comment by dwc0011** [comment]
@jurre Just wanted to see if there we are concerns with this.

**Comment by dwc0011** [comment]
bump

**Comment by lorengordon** [comment]
Wanted to check-in on this patch, anything else needed? We're currently blocked whenever codecommit is the source repository, and this patch seems to fix it.

File: common/lib/dependabot/metadata_finders/base/release_finder.rb [hunk 1]
```diff
@@ -195,7 +195,7 @@ def fetch_dependency_releases
           case source.provider
           when "github" then fetch_github_releases
           # Bitbucket doesn't support releases and Azure can't list API for annotated tags
-          when "bitbucket", "azure" then []
+          when "bitbucket", "azure", "codecommit" then []
```
Comment (jeffwidman) [inline common/lib/dependabot/metadata_finders/base/release_finder.rb:198]: Curious: does `codecommit` support releases?

A no-op is fine for now, but just wondering if this should have a TODO comment of adding future support for code commit releases, or if it's truly somethign that should be permanently a no-op... if the latter, than perhaps the comment for bitbucket/azure should also explain why codecommit was added to the no-op list.

**Review by pavera — APPROVED** [review]
Thanks @dwc0011!

File: common/lib/dependabot/metadata_finders/base/release_finder.rb (hunk 1 above)
Comment (dwc0011) [reply common/lib/dependabot/metadata_finders/base/release_finder.rb:198]: Codecommit does not support releases, there is no api, so the comment can be updated to reflect that as Bitbucket and Azure do.  
I am hesitant to update it though and cause the PR to need to be approved again, it took quite a while to get approved this go around and would hate for it to be moved to the back of the line again for a comment.

Comment (pavera) [reply common/lib/dependabot/metadata_finders/base/release_finder.rb:198]: I'm working on getting this merged/released today, so if you can update the comment, I'll get CI passed and get it merged in ASAP

Comment (dwc0011) [reply common/lib/dependabot/metadata_finders/base/release_finder.rb:199]: Done! Thank you very much!

**Pull request merged**